src/commands/openapi/upload.ts

```diff
@@ -170,9 +221,21 @@ export default class OpenAPIUploadCommand extends BaseCommand<typeof OpenAPIUplo
 
     if (this.flags['legacy-id']) {
       if (!matchingAPIDefinition) {
-        throw new Error(`No API definition found with legacy ID ${this.flags['legacy-id']}.`);
+        let existingDefinitionSuggestion = ' Did you mean to use the `--slug` flag instead?';
+        if (existingAPIDefinitions.length === 1) {
+          existingDefinitionSuggestion = ` 1 file was found — try replacing \`--legacy-id ${this.flags['legacy-id']}\` with \`--slug ${existingAPIDefinitions[0].filename}\`.`;
+        } else if (existingAPIDefinitions.length > 1) {
+          existingDefinitionSuggestion = ` ${existingAPIDefinitions.length} file(s) were found — try passing one of them via the \`--slug\` flag instead: ${filenames}.`;
+        }
```

Comment:
Not really blocking, but this behavior feels a little odd to me:
 - Person uploads a spec with `--legacy-id [ID-lookin-thing]`
 - We don't find a matching spec to that ID
 - If no specs exist, we suggest they upload using `--slug=[ID-lookin-thing]` ?
 - If spec(s) are found, we suggest they overwrite a different file with an unrelated slug

Because we don't know if any of the existing specs are actually the same one or not, so it feels a little strong to suggest that they try one of those slugs. Idk what I'd do instead though 😰